sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -353,11 +353,19 @@ class SQLListener(conf: SparkConf) extends SparkListener with Logging {
         }
 
         val driverUpdates = executionUIData.driverAccumUpdates.toSeq
-        val totalUpdates = (accumulatorUpdates ++ driverUpdates).filter {
-          case (id, _) => executionUIData.accumulatorMetrics.contains(id)
+        // filter out updates that aren't Longs
+        val totalUpdates = (accumulatorUpdates ++ driverUpdates).filter { update =>
+          update._2.isInstanceOf[Long]
         }
+        // Hack alert!
+        // metrics may be missing from the executionUIData because metrics are taken from the
+        // SparkPlan passed to withNewExecutionId, but some nodes in that plan link in LogicalPlan
+        // nodes that don't expose the actual SparkPlan or metrics because the SparkPlan is built
```

Comment:
I think it's better to have the metrics than to discard them. Why discard metrics if we know that they can be safely summarized and used?